src/sequential/sequential.hpp

```diff
@@ -0,0 +1,198 @@
+////////////////////////////////////////////////////////////////////////////////
+
+/// N.B.  I use int32_t for the key and value so that they fit into a 64-bit word
+///       if I so choose to do that. This way, we can do atomic updates. The
+///       extra metadata is a bit of an oof though.
+/// N.B.  I do not use *_t because this is a reserved name in POSIX.
+/// N.B.  I also named them something unique so it's easy to find/replace.
+typedef int32_t KeyType;
+typedef int32_t ValueType;
+/// N.B.  I use 'HashCode' because I want to distinguish 'hash' (verb) and
+///       'hash' (noun). Thus, I use 'hash code' to denote the latter.
+typedef size_t HashCodeType;
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// LOGGING MACROS (not thread safe)
+////////////////////////////////////////////////////////////////////////////////
+
+/// Create my own sketchy logger
+#define LOG_LEVEL_TRACE   6
+#define LOG_LEVEL_DEBUG   5
+#define LOG_LEVEL_INFO    4
+#define LOG_LEVEL_WARN    3
+#define LOG_LEVEL_ERROR   2
+#define LOG_LEVEL_FATAL   1
+#define LOG_LEVEL_OFF     0
+
+#define LOG_LEVEL LOG_LEVEL_DEBUG
+
+#define LOG_TRACE(x)  do { if (LOG_LEVEL >= LOG_LEVEL_TRACE)  { std::cout << "[TRACE]\t[" << __FILE__ << ":" << __LINE__ << "]\t[" << __FUNCTION__ << "]\t" << x << std::endl; } } while (0)
+#define LOG_DEBUG(x)  do { if (LOG_LEVEL >= LOG_LEVEL_DEBUG)  { std::cout << "[DEBUG]\t[" << __FILE__ << ":" << __LINE__ << "]\t[" << __FUNCTION__ << "]\t" << x << std::endl; } } while (0)
+#define LOG_INFO(x)   do { if (LOG_LEVEL >= LOG_LEVEL_INFO)   { std::cout << "[INFO]\t["  << __FILE__ << ":" << __LINE__ << "]\t[" << __FUNCTION__ << "]\t" << x << std::endl; } } while (0)
+#define LOG_WARN(x)   do { if (LOG_LEVEL >= LOG_LEVEL_WARN)   { std::cout << "[WARN]\t["  << __FILE__ << ":" << __LINE__ << "]\t[" << __FUNCTION__ << "]\t" << x << std::endl; } } while (0)
+#define LOG_ERROR(x)  do { if (LOG_LEVEL >= LOG_LEVEL_ERROR)  { std::cout << "[ERROR]\t[" << __FILE__ << ":" << __LINE__ << "]\t[" << __FUNCTION__ << "]\t" << x << std::endl; } } while (0)
+#define LOG_FATAL(x)  do { if (LOG_LEVEL >= LOG_LEVEL_FATAL)  { std::cout << "[FATAL]\t[" << __FILE__ << ":" << __LINE__ << "]\t[" << __FUNCTION__ << "]\t" << x << std::endl; } } while (0)
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// HELPER CLASSES
+////////////////////////////////////////////////////////////////////////////////
+
+enum SearchStatus {
+  FOUND_MATCH,
+  FOUND_SWAP,
+  FOUND_HOLE,
+  FOUND_NOHOLE,
+};
+
+/// @brief  Bucket for the Robin Hood hash table.
+struct SequentialBucket {
+  KeyType key;
+  ValueType value;
+  HashCodeType hashcode;
+  // A value of SIZE_MAX means that the bucket is empty. I do this hack so that
+  // we can fit this bucket into 4 words, which is more amenable to the hardware.
+  // A value of SIZE_MAX would be attrocious for performance anyways.
+  size_t offset;
```

Comment:
```suggestion
  KeyType key = 0;
  ValueType value = 0;
  HashCodeType hashcode = 0;
  // A value of SIZE_MAX means that the bucket is empty. I do this hack so that
  // we can fit this bucket into 4 words, which is more amenable to the hardware.
  // A value of SIZE_MAX would be attrocious for performance anyways.
  size_t offset = SIZE_MAX;
```
Generally better to use default-initialisation than a default constructor - especially for POD types, since defining a constructor removes the ability to use aggregate initialisation and other stuff you get for free like `std::swap`.